### ISSUE TITLE
App crashes with network connection

### ISSUE THREAD

**Issue by Gaaarz** [signalapp/Signal-iOS#5300]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description

 Old bug that returned. The old bug is #5105

### Steps to reproduce

- Open Signal while you are on network
- wait 0,5 seconds
- it crashes

### Device info

**Device**: Iphone 11 Pro

**iOS version**: 14.3

**Signal version:** 5.33.0 and 5.33.1

### Link to debug log

Already sent via email

**Comment by waikchiu5** [comment]
I got this last year and resumed after update the application.
However, this time cannot resume by update the newer version of application.

As a developer, I think that defect is relating to Desktop and IOS.
I mainly use Signal Desktop to chat with others and groups. As I know, IOS Signal has some issue about delay msg. You need to launch the app and receive the newer msg. 

If you interact many msg(>100) with others in Desktop. You will get app crash when launch the IOS Signal.
Like migration issue between desktop and IOS.

**Comment by Gaaarz** [comment]
What I also notice is, that Signal running full speed after crash and the process can't be terminated and wasting battery now. Sadly I have since 5.33.0 a useless app with battery functionality :)

Funny fun fact, if I share a picture from photo libary via Signal it dont crash and send it successfully. 

![image](https://user-images.githubusercontent.com/102361550/160578611-3da775c5-cc7a-4392-a457-7dc52be51de1.png)

**Comment by waikchiu5** [comment]
> 

I cannot fix this and waiting the solution. 
I am suing Desktop version to chat right now. The IOS version cannot open anymore

**Comment by precise-penguin** [comment]
Current iOS – same problem
Device: iPhone SE 2020
iOS version: 15.4
Signal version: 5.33.1

**Comment by peteygao** [comment]
Adding my report as well:
iPhone 13 Pro 1TB
iOS 15.4
Signal 5.33.0.25

**Comment by MysticalGrape** [comment]
Same issue. Fully patched iOS on iPhone 13 Pro with Signal 5.33.1. 
The app didn't crash when I enabled airplane mode. When connected then it crashes within 1-2 seconds. Reinstalling the app helps, but that involves relinking other devices and other inconveniences. Desktop app continues to work fine.

**Comment by Gaaarz** [comment]
> Same issue. Fully patched iOS on iPhone 13 Pro with Signal 5.33.1. 
> 
> The app didn't crash when I enabled airplane mode. When connected then it crashes within 1-2 seconds. Reinstalling the app helps, but that involves relinking other devices and other inconveniences. Desktop app continues to work fine. 
> 
> 
> 
> edit:
> 
> Same issue on linked iPad with Signal 5.33.1. The issue seemed to start right after sending voice message to existing chat from Signal Desktop 5.37.0 (on Macbook Air M1)

Did you loose your chat history if you delete your app and reinstall it? Probably yes, or?

**Comment by talkhunat** [comment]
> 

This is exactly what happened for me also, sent audio file to an existing chat and right after that mobile started crasing.
My device is iPhone 12 Mini, iOS 15.4, signal version 5.33.1 and signal desktop is 5.37.0

**Comment by MysticalGrape** [comment]
> > Same issue. Fully patched iOS on iPhone 13 Pro with Signal 5.33.1.
> > The app didn't crash when I enabled airplane mode. When connected then it crashes within 1-2 seconds. Reinstalling the app helps, but that involves relinking other devices and other inconveniences. Desktop app continues to work fine.
> > edit:
> > Same issue on linked iPad with Signal 5.33.1. The issue seemed to start right after sending voice message to existing chat from Signal Desktop 5.37.0 (on Macbook Air M1)
> 
> Did you loose your chat history if you delete your app and reinstall it? Probably yes, or?

I lost chat history on iPad and iPhone due to reinstall of Signal app. History on Macbook and Linux desktop survived after re-linking them to Signal.

**Comment by Gaaarz** [comment]
> > > Same issue. Fully patched iOS on iPhone 13 Pro with Signal 5.33.1.
> 
> > > The app didn't crash when I enabled airplane mode. When connected then it crashes within 1-2 seconds. Reinstalling the app helps, but that involves relinking other devices and other inconveniences. Desktop app continues to work fine.
> 
> > > edit:
> 
> > > Same issue on linked iPad with Signal 5.33.1. The issue seemed to start right after sending voice message to existing chat from Signal Desktop 5.37.0 (on Macbook Air M1)
> 
> > 
> 
> > Did you loose your chat history if you delete your app and reinstall it? Probably yes, or?
> 
> 
> 
> I lost chat history on iPad and iPhone due to reinstall of Signal app. History on Macbook and Linux desktop survived after re-linking them to Signal.
> 
> 
> 
> 
> 
> Edit: Signal app is still crashing on my iPhone and iPad, but I am able to share a picture taken on mobile from Camera app to  Signal app and it appears on linked Signal Desktop.

I can confirm as well, that sharing is working well.

**Comment by MysticalGrape** [comment]
Reinstalled the Signal app, relinked all other devices and things are stable so far. Recorded voice message from Signal Desktop to myself (Note to Self contact) and it got correctly synced to all devices and didn't crash iPhone and iPad apps. Let's see how it goes. I will post an update here when the issue will happen again.

**Comment by Gaaarz** [comment]
> Reinstalled the Signal app, relinked all other devices and things are stable so far. Recorded voice message from Signal Desktop to myself (Note to Self contact) and it got correctly synced to all devices and didn't crash iPhone and iPad apps. Let's see how it goes. I will post an update here when the issue will happen again. 

Maybe this bug appears if you send a voice message with an outdated desktop client.

**Comment by moalo** [comment]
Reporting same problem:

iPhone 13 Mini
iOS: 15.4.1
Signal: 5.33.1

MacBook Pro
macOS: 12.3
Signal: 5.37.0 production (M1)

**Comment by Flamity** [comment]
Same issue. Haven't been able to load signal since I updated. 

iPhone XS
iOS 15.4 (19E241)
Signal 5.33.1

**Comment by peteygao** [comment]
I just updated to Signal 5.33.2.2 on the iOS App Store and the issue is resolved! 👏

**Comment by waikchiu5** [comment]
Able to open the app after updated the latest version
But not sure it is completely resolve.

**Comment by kaktusztea** [comment]
This commit looks like that fixed it most probably:
https://github.com/signalapp/Signal-iOS/commit/db64c160aca1ba2866fa502236d831deca13ab70

**Comment by Gaaarz** [comment]
I can confirm, that the new update fixed the issue. Great work guys 💪🏻

**Comment by talkhunat** [comment]
> > Reinstalled the Signal app, relinked all other devices and things are stable so far. Recorded voice message from Signal Desktop to myself (Note to Self contact) and it got correctly synced to all devices and didn't crash iPhone and iPad apps. Let's see how it goes. I will post an update here when the issue will happen again. 
> 
> Maybe this bug appears only if you send a voice message with an outdated desktop client. 

In my case I'm 99% sure I sent the audiofile before updating desktop client. I got mobile working again by reinstalling signal, but its quite inconvenient way, since I lost all of my chathistories.

**Comment by jackreid** [comment]
I just started experiencing, if not the same bug, at least one that manifests the same symptoms. This is Signal v5.44.3 on Android 10. I can't directly submit the debugging logs as the app crashes too quickly once I turn off airplane mode. I have tried to grab some relevant screenshots. Unfortunately I can't get the logs from the exact moment of startup (around 15:05:12) because when I scroll up in the logs, it will suddenly jump to a few minutes earlier (15:02:36) and won't allow me to scroll back down. Not sure if that is related to the bug or independent.

I did not send any voice messages.


![Screenshot_20220804-145827](https://user-images.githubusercontent.com/25727967/182934326-78f4d798-d5e8-4697-ae20-d87b0684a273.png)
![Screenshot_20220804-151002](https://user-images.githubusercontent.com/25727967/182934328-f1b1b6c4-72dc-46e7-91cb-4139be7f0e29.png)
![Screenshot_20220804-151010](https://user-images.githubusercontent.com/25727967/182934329-567232ea-7341-4063-a8a1-4b2d42a17902.png)
